### PR TITLE
PurriRoll

### PR DESCRIPTION
**Pull request by RichTeaMan**
# Description

Added a Dice Maiden Parser. Purrito now listens for dice rolls and reacts with a MEOW!!! is a dice roll was 20.

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tenatively.

### CODE REVIEW

File: src/DiceMaiden/DiceMaidenResult.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export class DiceMaidenResult {
+
+    success: boolean = false;
```
Comment (djaustin) [inline src/DiceMaiden/DiceMaidenResult.ts]: What does success mean in this context?

File: src/index.ts [hunk 2]
```diff
@@ -35,17 +37,28 @@ client.on('guildDelete', guild => {
 client.on('message', async (message: Message) => {
     // This event will run on every single message received, from any channel or DM.
 
-    // Ignore other bots
+    console.log(`On message fired - ${message.author.username}, ${message.content}.`)
+
+    if (message.author.username === 'Dice Maiden') {
```
[the remark this answers is not in the document]
Comment (RichTeaMan) [reply src/index.ts]: Username doesn't seem to change. Tag is the full ID with a hash, eg: Dice Maiden#9678

[the remark this answers is not in the document]
Comment (djaustin) [reply src/index.ts]: I think tag would be better as other users with the username Dice Maiden would get picked up with this

File: src/DiceMaiden/DiceMaidenResult.ts (hunk 1 above)
Comment (RichTeaMan) [reply src/DiceMaiden/DiceMaidenResult.ts]: I've added more detail.

File: src/DiceMaiden/ParseDiceMaiden.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    const result = new DiceMaidenResult();
+
+    const rollerName = messageContent.match(/\S+/)
+    if (rollerName !== null) {
+        result.rollerName = rollerName[0];
+    }
+
+    const totalResultStr = messageContent.match(/(?<=Result:\s)\d+/)
+    if (totalResultStr && totalResultStr.length === 1) {
+        const totalResult = Number.parseInt(totalResultStr[0], 10)
```
[the remark this answers is not in the document]
Comment (djaustin) [reply src/DiceMaiden/ParseDiceMaiden.ts]: It pains me to say but this one is till here :(

[the remark this answers is not in the document]
Comment (RichTeaMan) [reply src/DiceMaiden/ParseDiceMaiden.ts]: (╯°□°)╯︵ ┻━┻

**Pull request merged**